cypress-io/cypress
# fix: don't show deps warning for projects that don't require bundler

- Closes #23216

### User facing changelog
Remove misleading dependency warning in Launchpad for Next.js applications

### Additional details
We were including the bundler configured in the users `cypress.config` as a dependency to validate against. Next bundles it's own version of Webpack internally, meaning the user won't have a `node_modules/webpack` dep installed. The only dependencies we should validate against are the dependencies listed in the frameworks `dependencies` field.

For most frameworks, the `dependencies` list already includes a bundler. For projects that don't require `webpack/vite` to be explicitly installed (such as projects like Next.js that bundle it internally or `react-scripts` in which it is a transient dependency), it should not be listed. I've updated the list to remove the bundler from projects that had it listed when it shouldn't be.

### Steps to test
- Create a next application with `npx create-next-app@latest my-app` (don't put this project into a folder where webpack can be found up recursively...)
- cd into application and run `npm i -D cypress`
- Open cypress `npx cypress open`
- Walk through CT setup, browser screen should show `webpack` dep warning
- Kill Cypress and run local Cypress `yarn cypress:open`
- Add new next.js application and click on CT testing, no warning

### How has the user experience changed?

https://user-images.githubusercontent.com/25158820/187308042-0d025aaf-4975-43ad-9fd4-bc2ecc5b2ab5.mov



### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
